src/main/java/com/puppycrawl/tools/checkstyle/api/Configuration.java

```diff
@@ -62,4 +63,10 @@
      * @return unmodifiable map containing custom messages
      */
     ImmutableMap<String, String> getMessages();
+
+    /**
+     * Gets the thread mode settings.
+     * @return the thread mode settings.
+     */
+    ThreadModeSettings getThreadModeSettings();
```

Comment:
This is API, we should avoid this change and I am ok to even have some refrection(or unsafe casting) code to main code to get this settings.
It is easy to make changes in API and very hard to remove code from it.
We play with hiding new options in Main.java but do API changes in the same time .... smth is not good.
If we are still not sure, lets hide implementation so  no public information on it, no API changes.

@soon , @sabaka can we avoid this change ?